doc/neps/nep-0043-extensible-ufuncs.rst

```diff
@@ -0,0 +1,1261 @@
+the array when considering a ufunc with only a single input::
+
+    res = np.positive(arr)
+
+could be implemented (conceptionally) as:
+
+    positive_impl = arr.dtype.positive
+    res = positive_impl(arr)
+
+However, unlike methods, ``positive_impl`` is not stored on the dtype itself.
+It is rather the implementation of ``np.positive`` for a specific DType.
+Current NumPy partially exposes this "choice of implementation" using
+the ``dtype`` (or more exact ``signature``) attribute in universal functions,
+although these are rarely used:
+
+    np.positive(arr, dtype=np.float64)
+
+forces NumPy to use the ``positive_impl`` written specifically for the Float64
+DType.
+
+This NEP makes this distinction more explicit, by creating a new object to
+represent ``positive_impl``::
+
+    positive_impl = np.positive.resolve_impl(type(arr.dtype))
+
+While the creation of a ``positive_impl`` object and the ``resolve_impl``
+method is part of this NEP, the following code::
+
+    res = positive_impl(arr)
+
+may not be implemented initially and is not central to the redesign.
+
+In general NumPy universal functions can take many inputs.
+This requires looking up the implementation by considering all of them
+and makes ufuncs "multi-methods" with respect to the input DTypes::
+
+    add_impl = np.add.resolve_impl(type(arr1.dtype), type(arr2.dtype))
+
+This NEP defines how ``positive_impl`` and ``add_impl`` will be represented
+as a new ``ArrayMethod`` and can be defined outside of NumPy.
+Further, it defines how ``resolve_impl`` will be implemented, covering the
+dispatching and promotion.
+
+The reasons for this split may be made more clear in the section
+`Steps involved in a UFunc call`_.
+
+
+Defining a new ufunc implementation
+===================================
+
+An example of how to add a new loop, will look the following way;
+initially using a C-API:
+
+.. code-block:: python
+
+    class StringEquality(BoundArrayMethod):
+        nin = 1
+        nout = 1
+        DTypes = (String, String, Bool)
+
+        def resolve_descriptors(context, given_descrs):
+            """The strided loop supports all input string dtype instances
+            and always returns a boolean. (String is always native byte order.)
+
+            Defining this function is not necessary, since NumPy can provide
+            it by default.
+            """
+            assert isinstance(given_descrs[0], context.DTypes[0])
+            assert isinstance(given_descrs[1], context.DTypes[1])
+            
+            # The operation is always "safe" casting (most ufuncs are)
+            return (given_descrs[0], given_descrs[1], context.DTypes[2]()), "safe"
+
+        def strided_loop(context, n, data, strides):
+            # n: Number of 
+            # data: Pointers to the array data.
+            # strides: strides to iterate all elements
+            num_chars1 = context.descriptors[0].itemsize
+            num_chars2 = context.descriptors[0].itemsize
```

Comment:
I think you need dimension as well as strides. If dimensions are in the context, then strides could be as well. Also `# n: Number of` seems like a fragment.